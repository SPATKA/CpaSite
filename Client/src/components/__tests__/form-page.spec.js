import React from "react"
import renderer from "react-test-renderer"
import FormPage from "../formPage"
import { render, fireEvent } from "@testing-library/react";
// import * as utils from '@utils'

// utils.fetch = jest.fn().mockResolvedValue({
//   statusCode: 200,
//   body: JSON.stringify({
//     message: 'Mail sent successfully'
//   }),
// })
require('jest-fetch-mock').enableMocks()
fetchMock.dontMock()

describe("FormPage", () => {

  beforeEach(() => {
    
    fetchMock.mockIf(/^https?:\/\/3ct3mrc87g.execute-api.us-east-1.amazonaws.com.*$/,async req => {
      if (req.url.endsWith("/dev/sendMail")) {
        return Promise.resolve({
          statusCode: 200,
          body: JSON.stringify({
            message: 'Mail sent successfully'
          }),
        })
      } else {
        return {
          status: 404,
          body: "Not Found"
        }
      }
    })
  })

  it("renders correctly", () => {
    const tree = renderer
      .create(<FormPage />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  test("if no data inserted in form on submit then should display correct error message", () => {
    const { getByTestId, findByText } = render(<FormPage />);

    getByTestId("submit");

    fireEvent.click(getByTestId("submit"));

    findByText("Name is required.");
    findByText("Phone number is required.");
    findByText("Email address is required.");
  });

  test.skip("if valid data inserted in form then on submit should display message 'detail submited successfully", () => {
    const { getByTestId } = render(<FormPage />);

    getByTestId("submit");

    fireEvent.input(getByTestId("name"), {
      target: {
        value: 'testName'
      }
    });
    fireEvent.input(getByTestId("email"), {
      target: {
        value: 'test@mail.com'
      }
    });
    fireEvent.input(getByTestId("phone"), {
      target: {
        value: '9876543210'
      }
    });

    fireEvent.click(getByTestId("submit"))

    // await expect(getByTestId("submit-success")).toBeInTheDocument()
    expect(fetch).resolves.toBe({
      statusCode: 200,
      body: JSON.stringify({
        message: 'Mail sent successfully'
      })
    })
  });

  test.skip("on reset should reset the form value", () => {
    const { getByTestId, findByText } = render(<FormPage />);

    fireEvent.click(getByTestId("submit"));

    findByText("Name is required.");
    findByText("Phone number is required.");
    findByText("Email address is required.");

    fireEvent.click(getByTestId("reset"));

    expect(fireEvent.input(getByTestId("name")).valueOf()).toBe(true)
  });
});
