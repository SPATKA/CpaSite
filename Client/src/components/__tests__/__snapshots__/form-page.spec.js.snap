// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormPage renders correctly 1`] = `
<div
  className="mx-auto"
  style={
    Object {
      "maxWidth": "400px",
    }
  }
>
  <form
    method="post"
    onSubmit={[Function]}
  >
    <div
      className="mb-3"
    >
      <label
        className="form-label"
        htmlFor="name"
      >
        Name
      </label>
      <input
        aria-invalid="false"
        className="form-control"
        data-testid="name"
        id="name"
        name="name"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Enter your name"
        type="text"
      />
    </div>
    <div
      className="mb-3"
    >
      <label
        className="form-label"
        htmlFor="email"
      >
        Email address
      </label>
      <input
        aria-invalid="false"
        className="form-control"
        data-testid="email"
        id="email"
        name="email"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="your@email.address"
        type="email"
      />
    </div>
    <div
      className="mb-3"
    >
      <label
        className="form-label"
        htmlFor="phone"
      >
        Phone Number
      </label>
      <input
        aria-invalid="false"
        className="form-control"
        data-testid="phone"
        id="phone"
        name="phone"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Enter phone number"
        type="text"
      />
    </div>
    <div
      className="mb-3"
    >
      <label
        htmlFor="comments"
      >
        Comments
      </label>
      <textarea
        aria-invalid="false"
        className="form-control"
        id="comments"
        name="comments"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Leave a comment here"
      />
    </div>
    <div
      className="d-flex justify-content-end"
    >
      <button
        className="btn btn-secondary mx-1"
        data-testid="reset"
        onClick={[Function]}
        type="button"
      >
        reset
      </button>
      <button
        className="btn btn-primary"
        data-testid="submit"
        type="submit"
      >
        Send
      </button>
    </div>
  </form>
</div>
`;
